diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/SingleLineStatementsRule.kt

```diff
@@ -36,22 +37,40 @@ class SingleLineStatementsRule : Rule("statement") {
 
     private fun checkSemicolon(node: ASTNode) {
         node.getChildren(semicolonToken).forEach {
-            if (!isSemicolonInMidLine(it)) {
+            if (!isSemicolonAtEndOfLine(it)) {
                 if (isError(it)) {
                     MORE_THAN_ONE_STATEMENT_PER_LINE.warnAndFix(configRules, emitWarn, isFixMode, findWrongText(it),
                             it.startOffset) {
-                        node.addChild(PsiWhiteSpaceImpl("\n"), it)
-                        node.removeChild(it)
+                        if (it.treeParent.elementType == ENUM_ENTRY){
+                            node.treeParent.addChild(PsiWhiteSpaceImpl("\n"), node.treeNext)
+                        } else {
+                            if (!it.isBeginByNewline())
+                                node.addChild(PsiWhiteSpaceImpl("\n"), it)
```

Comment:
I just thought that this will lead to having two consecutive `WHITE_SPACE` nodes (the one with new newline and the one after removed `;`), which may cause issues when applying `IndentationRule` on transformed code. Might be hard to check now, so could you at least add fixme comment here, that white spaces should probably be merged here.